lib/chef/knife.rb

```diff
@@ -375,7 +375,7 @@ def merge_configs
     #   - :cli_default - came from a declared CLI `option`'s `default` value.
     #   - nil - if they key does not exist
     def config_source(key)
-      return :cli if @original_config.include? key
+      return :cli if config.include? key
```

Comment:
This test would always return true. `config` is the merged configuration, while `@original_config` is the pre-merge copy.